### PR TITLE
Keyframes cleanup

### PR DESCRIPTION
**Pull request by davidaurelio**
Most work done by @qfox

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -2,6 +2,8 @@
 * Added all instance properties of KeyframeAnimation to the prototype and documented them
 * Removed unused KeyframeAnimation#removeSubject and #removeSubjects
 * Fixed an issue with KeyframeAnimation where it would trip over certain undefined values
+* Fixed a bug in KeyframeAnimation#_fillInProperties where it would refer to the wrong instance property easingFn (-> easing)
```
Comment (basecode) [inline CHANGELOG]: Pls add line-breaks.

File: src/runner/animation/keyframe_animation.js [hunk 2]
```diff
@@ -391,12 +380,10 @@ define([
...
      */
     removeSubject: function(subject) {
-      for (var i = 0, l = this.subjects.length; i < l; ++i) {
-        if (this.subjects[i].subject === subject) {
-          this.subjects.splice(i, 1);
-          for (var a = 0, al = this.animations.length; a < al; ++a) {
-            this.animations[a].removeSubject(subject);
-          }
+      var subjects = this.subjects;
+      for (var i = subjects.length; i >= 0; i -= 1) {
```
Comment (basecode) [inline src/runner/animation/keyframe_animation.js:384]: I thought we agreed on using `tools.forEach` in "non-critical" code?

File: src/runner/animation/keyframe_animation.js [hunk 3]
```diff
@@ -408,7 +395,7 @@ define([
      * @return {KeyframeAnimation}
      */
     removeSubjects: function(subjects) {
-      forEach(subjects, tools.hitch(this, 'removeSubject'));
+      forEach(subjects, this.removeSubject, this);
```
Comment (basecode) [inline src/runner/animation/keyframe_animation.js:398]: nice

File: src/runner/animation/keyframe_animation.js (hunk 2 above)
Comment (basecode) [inline src/runner/animation/keyframe_animation.js:384]: Doesn't work with splice.. nevermind.

File: src/runner/animation/keyframe_animation.js [hunk 4]
```diff
@@ -472,7 +459,7 @@ define([
      */
     _fillInProperties: function(initialValues) {
 
-      var easingFn = this.easingFn,
+      var easingFn = this.easing,
           lastFrame = this.duration,
           keys = this.keys,
```
Comment (basecode) [inline src/runner/animation/keyframe_animation.js:464]: Rename `easingFn` to `easing` and `lastFrame` to `duration` please.

File: CHANGELOG [hunk 5]
```diff
@@ -2,6 +2,8 @@
...
+* Fixed a bug in setting the KeyframeAnimation#delay due to flawed logic condition
```
Comment (basecode) [inline CHANGELOG]: Remove this line please.

File: src/runner/animation/keyframe_animation.js (hunk 2 above)
Comment (davidaurelio) [inline src/runner/animation/keyframe_animation.js:384]: Also does not work in reverse order. `filter()` would do the trick here.

Comment (basecode) [inline src/runner/animation/keyframe_animation.js:384]: If reverse order is important here, we should add a comment.

**Pull request merged**